### ISSUE TITLE
Question about project state.

### ISSUE THREAD

**Issue by emilHof** [max397574/rust-norg#3]
Heyy all!

Hopefully this is not a silly question and if it is, feel free to ignore haha!

I was wondering what the state of this project was in terms of
1. the development status
2. its current usability
3. and remaining TODO's.

Asking this is out of plain curiosity and interest in contributing, yet totally feel free to ignore if busy! :D

**Comment by max397574** [comment]
I'll be completely honest I don't remember what needs to be done and currently I don't have the time to work on this project. If I'd work on it again I would probably redo it because since this I got a lot better with rust and also wrote another parser.
If you'd like to contribute feel free to do so, I'd be happy to merge a pr. I just want to warn you that perhaps this project will eventually be superseeded by a new one.

**Comment by emilHof** [comment]
hi @max397574 !

oh i see, that makes sense! would you likely start over with the current repo or do it completely from scratch with a new version managed remote ?

**Comment by max397574** [comment]
Well what I want to do in the future is an lsp for norg
and for that I'd either write my own parser (likely with a library like nom) or use tree-sitter
I could also use a "manual" version like here but that's just a lot more work I guess. I haven't decided yet
this will be a completely new project but there's a chance I can reuse big parts of this one

**Comment by emilHof** [comment]
ohh wow! that is a super cool idea! is there already such a thing like an LSP for norg out there or an ongoing project in another language, or would this be completely novel?

**Comment by max397574** [comment]
this would be completely novel
I think there was a project somewhere in zig but it's abandoned

**Comment by emilHof** [comment]
ooo, that would be incredibly cool then! are you planning on making it a fully fledged LSP with hints, autocomplete, and the likes ? :D

**Comment by max397574** [comment]
here it is
https://github.com/nvim-neorg/neorg-lsp

**Comment by max397574** [comment]
yes I defnitely want autocomplete for task-states and filenames in links perhaps syntax hints on hover
I'm sure there will be more thing we could do but those are the ones I currently have in mind

**Comment by emilHof** [comment]
ahh! is that the zig version?

**Comment by max397574** [comment]
yes
